lib/Widgets/DynamicNotebook.vala

```diff
@@ -1184,18 +1194,23 @@ namespace Granite.Widgets {
             }
         }
 
-        private void on_close_others (Tab tab) {
-            var num = 0; //save num, in case a tab refused to close so we don't end up in an infinite loop
-
-            for (var j = 0; j < tabs.length (); j++) {
-                if (tab != tabs.nth_data (j)) {
-                    tabs.nth_data (j).closed ();
-                    if (num == n_tabs) break;
-                    j--;
+        private void on_close_others (Tab clickedTab) {
+            tabs.copy ().foreach ((tab) => {
+                if (tab != clickedTab) {
+                    tab.closed ();
                 }
+            });
+        }
 
-                num = n_tabs;
-            }
+        private void on_close_others_right (Tab clickedTab) {
+            var toTheRight = false; 
+
+            tabs.copy ().foreach ((tab) => {
+                if (toTheRight)
```

Comment:
One-line `if` statements should also have curly braces: https://elementary.io/docs/code/reference#indentation

Reply:
done